src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -219,6 +219,11 @@ private[health] class HealthCheckActor(
     val newHealth = instanceHealth.newHealth
 
     logger.info(s"Received health result for app [${app.id}] version [${app.version}]: [$result]")
+
+    val healthOfInstanceId = healthByTaskId.find(_._1.instanceId == instanceId)
+    if (healthOfInstanceId.isDefined)
+      healthByTaskId.remove(healthOfInstanceId.get._1)
```

Comment:
Hmm, can you detail how this fixes the bug please?

When I read it naively, I understand that you remove healthResults from current instance? Shouldn't it be a `!=` instead to remove non matching health results?

Reply:
Done. Thanks.